extensions/hibernate-search-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/elasticsearch/runtime/HibernateSearchElasticsearchRuntimeConfig.java

```diff
@@ -118,6 +119,12 @@
         @ConfigItem
         DiscoveryConfig discovery;
 
+        /**
+         * Configuration for the thread pool assigned to the backend.
```

Comment:
Could you add a short paragraph to describe when one might want to tune this?

Reply:
Honestly, for the Elasticsearch backend, I'm not sure. Everything happening in these threads is non-blocking, except maybe some logging than can happen on timeouts/errors. So, theoretically, having more threads than the number of available cores is pointless.

I suppose one could want to set it to less than the number of cores, just to have less threads around?

Like all these performance settings, they are mostly here so that users are free to experiment. I don't have the answer to all performance problems, unfortunately.

Reply:
In that case it might be wise to not expose anything?  Remember in Quarkus we're trying to minimise the configuration needs.

I would suggest to defer exposing this to when we have a better understanding for it. For example, I suppose it's possible such an exploration concludes there's only need for some "profile" enums: "light" (default) vs "throughput" (matching num cores).

Reply:
I came up with a reason to use this setting a few minutes after I wrote this comment, and I documented it in a following commit: you may want to set this to a low number if your machine has many cores and you only have one index with one queue, for example.

While I'm all for this approach of "not exposing until we're sure" for feature-related settings, I'm not sure it makes sense when it comes to performance settings which will likely be different for every machine, and for which we don't have a silver bullet. In you example, I'd have a really hard time to find the number of core to assign to the "light" profile.

Reply:
yes I'm skeptical about applying such strategy to performance levers as well - but we're asked to try hard, so here I am having you do this exercise of defending your flags :)

I suppose one thing we could try to improve on is ergonomic strategies: rather than exposing such options, ideally we'd study our own statistics at runtime and automate what you'd do as a smart human who knows the project well. Essentially: ask yourself which stats would you need to know to make this perform better? Then what would you do? And automate that - sometimes this still needs some configuration flags, but they are more likely to look like "profiles" and strategy names, so that people can at least express what they are optimising for.

I'm sure that won't work with all options, but it makes sense when there's a possibility of automation to avoid exposing options.

(That being sad - beyond the exercise - I'm fine to keep this configuration property, thanks for your explanations)